examples/motoko_examples/basic-dao/src/basic_dao.ts

```diff
@@ -0,0 +1,385 @@
+  accounts = accounts_fromArray(initAccounts);
+  proposals = proposals_fromArray(initProposals);
+  nextProposalId = 0n;
+  systemParams = initSystemParams;
+}
+
+export function* heartbeat(): Heartbeat {
+  // We have to inline executeAcceptedProposals function here because of a limitaion of azle for now
+  const allProposals = Object.keys(proposals).map((id) => proposals[id]);
+
+  const acceptedProposals = allProposals.filter((p) => {
+    return 'accepted' in p.state;
+  });
+  // Update proposal state, so that it won't be picked up by the next heartbeat
+  for (let proposal of acceptedProposals) {
+    updateProposalState(proposal, { executing: null });
+  }
+
+  for (let proposal of acceptedProposals) {
+    try {
+      const payload = proposal.payload;
+
+      let res = yield ic.call_raw(
+        payload.canister_id,
+        payload.method,
+        payload.message,
+        0n
+      );
+
+      if ('ok' in res) {
+        updateProposalState(proposal, { succeeded: null });
+      } else {
+        updateProposalState(proposal, { failed: res.err });
+      }
+    } catch (e) {
+      updateProposalState(proposal, { failed: 'Execution failed' });
+    }
+  }
+}
+
+function accountGet(id: Principal): Opt<Tokens> {
+  return accounts[id] ?? null;
+}
+
+function account_put(id: Principal, tokens: Tokens) {
+  accounts[id] = tokens;
+}
+
+function proposalGet(id: nat): Opt<Proposal> {
+  return proposals[id.toString()] ?? null;
+}
+
+function proposalPut(id: nat, proposal: Proposal) {
+  proposals[id.toString()] = proposal;
+}
+
+// Transfer tokens from the caller's account to another account
+export function transfer(transfer: TransferArgs): Update<TransferResult> {
+  const caller = ic.caller();
+  const from_tokens = accountGet(caller);
+  switch (from_tokens) {
+    case null: {
+      return { err: 'Caller needs an account to transfer funds' };
+    }
+    case from_tokens as Tokens: {
+      const fee: nat = systemParams.transfer_fee.amount_e8s;
+      const amount = transfer.amount.amount_e8s;
+      if (from_tokens.amount_e8s < amount + fee) {
+        return {
+          err: `Caller's account has insufficient funds to transfer ${transfer.amount}`,
+        };
+      } else {
+        const from_amount: nat = from_tokens.amount_e8s - amount - fee;
+        account_put(caller, { amount_e8s: from_amount });
+        const to_amount =
+          (accountGet(transfer.to) ?? zeroToken).amount_e8s + amount;
+        account_put(transfer.to, { amount_e8s: to_amount });
+        return { ok: 0n };
+      }
+    }
+    default: {
+      return { err: 'Unexpected error' };
+    }
+  }
+}
+
+/// Return the account balance of the caller
+export function accountBalance(): Query<Tokens> {
+  return accountGet(ic.caller()) ?? zeroToken;
+}
+
+// Lists all accounts
+export function listAccounts(): Query<Account[]> {
+  const accountList = Object.keys(accounts).map((owner) => ({
+    owner,
+    tokens: accounts[owner],
+  }));
```

Comment:
Is there a reason for the extra set of parentheses around the body of the lambda?

Reply:
Oh I see, it's shorthand.